test/orm/declarative/test_typed_mapping.py

```diff
@@ -872,9 +874,17 @@ class User(Base):
             reverse_u_optional_data: Mapped[
                 Union[Decimal, float, None]
             ] = mapped_column()
+
             float_data: Mapped[float] = mapped_column()
             decimal_data: Mapped[Decimal] = mapped_column()
 
+            if compat.py310:
+                pep604_data: Mapped["float | Decimal"] = mapped_column()
+                pep604_reverse: Mapped["Decimal | float"] = mapped_column()
```

Comment:
can we also try with non string form? Like `Mapped[Decimal | float]`? Or there are already tests like this?

Reply:
Yep, absolutely. Somehow I'd convinced myself that the forward refs were necessary for the syntax to parse on older py vers.. :facepalm: The guards are enough.

I've tested both cases.